jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -38,17 +42,38 @@
 
 /** Builds a model {@link Image}. */
 class BuildImageStep
-    implements AsyncStep<AsyncStep<Image<CachedLayer>>>, Callable<AsyncStep<Image<CachedLayer>>> {
+    implements AsyncStep<AsyncStep<Image<Layer>>>, Callable<AsyncStep<Image<Layer>>> {
 
   private static final String DESCRIPTION = "Building container configuration";
 
+  @VisibleForTesting
+  static Layer cacheEntryToLayer(CacheEntry cacheEntry) {
```

Comment:
What if `CacheEntry` implements (extends) `Layer`?

Reply:
That's a good idea. I'll file a separate PR for that: #1055